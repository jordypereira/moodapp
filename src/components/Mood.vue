<template>
  <button>{{ title }}</button>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "Mood",
  props: {
    title: String,
  },
});
</script>
