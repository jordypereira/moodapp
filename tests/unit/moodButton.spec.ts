import { shallowMount,  } from '@vue/test-utils'
import Mood from '@/components/Mood.vue'

describe('Mood.vue', () => {
  it('renders props.title when passed', () => {
    const title = 'new message'
    const wrapper = shallowMount(Mood, {
      props: { title }
    })
    expect(wrapper.text()).toMatch(title)
  })
  it('Doesnt change style', () => {
    const title = 'new message'
    const wrapper = shallowMount(Mood, {
      props: { title }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
