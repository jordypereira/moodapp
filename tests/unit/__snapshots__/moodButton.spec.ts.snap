// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mood.vue Doesnt change style 1`] = `
<button>
  new message
</button>
`;
